main.go

```diff
@@ -111,9 +112,16 @@ func (c *Controller) handleNewCSR(key string) error {
 		return nil
 	}
 
-	if pending := recentlyPendingCSRs(c.indexer); pending > maxPendingCSRs {
-		klog.Infof("ignoring all CSRs as too many recent pending CSRs seen: %d", pending)
-		return nil
+	machines, err := c.machines.List(metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list machines: %v", err)
+	}
+
+	maxPending := getMaxPending(machines.Items)
+
+	if pending := recentlyPendingCSRs(c.indexer); pending > maxPending {
+		klog.Errorf("Pending CSRs: %d; Max pending allowed: %d", pending, maxPending)
+		return fmt.Errorf("Difference between pending CSRs and machines > 100: %v", pending)
```

Comment:
@michaelgugino we should preserve the return nil here so upcoming CSRs are ignored